content/src/ports/gzipCompressor.ts

```diff
@@ -0,0 +1,68 @@
+import destroy from 'destroy'
+import * as path from 'path'
+import { pipeline } from 'stream'
+import { promisify } from 'util'
+import { createGzip, createUnzip } from 'zlib'
+import { AppComponents } from '../types'
+const pipe = promisify(pipeline)
+
+export type CompressionResult = {
+  originalSize: number
+  compressedSize: number
+}
+
+export interface FileCompressor {
+  compress(sourceFilepath: string, destinationFilepath: string): Promise<CompressionResult>
+  decompress(sourceFilepath: string, destinationFilepath: string): Promise<boolean>
+}
+
+export function createGzipCompressor(components: Pick<AppComponents, 'logs' | 'fs'>): FileCompressor {
+  return {
+    async compress(sourceFilepath: string, destinationFilepath: string) {
```

Comment:
Since the compressor doesn't hold any state or lifecycle, I recommend you to move these functions to the `logic` folder and receiving the components as argument. If possible, prevent creating components that don't hold state or lifecycle hooks

```ts
export async function compress(components: Pick<AppComponents,'fs'>, sourceFilepath: string, destinationFilepath: string)
```